kiva/pdf.py

```diff
@@ -621,7 +624,18 @@ def set_font(self, font):
         face_name = font.face_name
         if face_name == "":
             face_name = "Helvetica"
-        self.gc.setFont(face_name, font.size)
+
+        try:
+            self.gc.setFont(face_name, font.size)
+        except KeyError:
+            # Face name is not recognized.
+            # Register the TTF font that kiva's font_manager located
+            filename = font.findfont()
+            if not os.path.splitext(filename)[-1] in (".ttf", ".ttc"):
+                raise
+
+            pdfmetrics.registerFont(TTFont(face_name, filename))
```

Comment:
As far as I can tell, this embeds the font in the PDF file. I'm not sure if we want that or not.

PDF has a set of "safe" fonts which come with Acrobat Reader. I think maybe what we want to do in this method is try to map all requested fonts to the nearest match in this list.

Reply:
> As far as I can tell, this embeds the font in the PDF file.

Based off the link below, I believe this is true as well.

> PDF has a set of "safe" fonts which come with Acrobat Reader. I think maybe what we want to do in this method is try to map all requested fonts to the nearest match in this list.

I am unsure about this / why it would be necessary, but my understanding of all this is very much not complete / still growing.  When would embedding the font in the PDF file not work or be problematic?

ref: https://www.reportlab.com/docs/reportlab-userguide.pdf
pg. 28 and 49-52

Reply:
> When would embedding the font in the PDF file not work or be problematic?

You're right. It's probably OK.